### ISSUE TITLE
Detailed SMART info in disk details

### ISSUE THREAD

**Issue by NanoSector** [cockpit-project/cockpit#15010]
Page: Storage -> Disk

Cockpit currently shows just whether a disk is OK or not along its temperature.

I have a drive in my system with reallocated sectors and logged SMART errors, of which `smartd` continuously spews errors in the log. Naturally I'd like to replace the drive, but I have to resort to a terminal window in order to gather details about my drive. It would be nice if Cockpit showed me more than the drive being "OK" (which it obviously is not).

GNOME Disks has a nice implementation of this, maybe Cockpit could do something similar:
![image](https://user-images.githubusercontent.com/1280380/101240531-ef9c1780-36ef-11eb-8d4f-aa40efedb3c7.png)

**Comment by marusak** [comment]
Indeed, but we have this already reported: https://github.com/cockpit-project/cockpit/issues/1134

**Comment by garrett** [comment]
This is about showing SMART information, not necessarily errors. I think we should have both. Showing smart information belongs in the disk details on the storage page.

It could be implemented in steps:

1. Show high-level status on the disk details page
2. Have a link for more details, which has a sub-page for the link _or_ an expanded section (progressive disclosure) with the SMART information.
3. Have an action to re-trigger SMART scanning (quick/detailed)

**Comment by garrett** [comment]
This probably needs design. I'll be happy to do it as soon as a developer's assigned to it or has interest in it... or whenever we add it to our schedule.

**Comment by tonyhardcode** [comment]
Would be awesome to view the SMART info on the main Overview page as well 👍

**Comment by jelly** [comment]
There is a PR open https://github.com/cockpit-project/cockpit/pull/19103 for this feature, it's mostly JavaScript as we get the SMART [data from udisks over dbus](http://storaged.org/doc/udisks2-api/latest/gdbus-org.freedesktop.UDisks2.Drive.Ata.html).

**Comment by mlincett** [comment]
While the progress on SMART reporting is great so far, accessing the full SMART report is still not possible. Can this issue be reopened or should we rather open a new one? @jelly

**Comment by jelly** [comment]
> While the progress on SMART reporting is great so far, accessing the full SMART report is still not possible. Can this issue be reopened or should we rather open a new one? [@jelly](https://github.com/jelly)

Feel free to open a new one but please specify which exact SMART details you want to see, we are limited to what is available in [udisks](https://storaged.org/doc/udisks2-api/latest/gdbus-org.freedesktop.UDisks2.Drive.Ata.html)

**Comment by madejoyful** [comment]
Question (from a Linux newbie): Is it possible to extract the values of the SMART test to a file or email? It would be very cool, to send a weekly report to my mail address. I can pull the data from a file or whatever to send it, but need to know where the information is stored.
